### DISCUSSION TITLE
help debugging | java agent | db spans, http spans not part of same trace

### DISCUSSION THREAD

**Discussion by someone-1** [apache/skywalking#10430] [Q&A] [answered]
i was integrating skywalking java agent with play application 
this application is using h2 db  https://github.com/playframework/play-samples/tree/2.8.x/play-java-rest-api-example/app/v1/post 

the span operations of the db are not appearing in the same trace as the http requests. they are in different trace  
i need help in debugging the agent.

i added breakpoint to 
https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/h2-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdbc/h2/PooledJdbcConnectionConstructorInterceptor.java#L27
and 
https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/h2-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdbc/h2/JdbcXAConnectionConstructorInterceptor.java#L33

but they are not getting invoked at run time 
unable to find where in h2 plugin the spans are created

pointing in right direction is very helpful

**Comment by wu-sheng** [comment, marked as answer]
In another trace, it means the database relative codes run in another thread. Such as thread-1 is calling the target of interceptors, then it switches to another thread-2, and run JDBC relative codes.

**Reply by someone-1** [reply]
got it, thinking of linking them as operation in the the same trace, makes debugging easier.

**Reply by wu-sheng** [reply]
If you want to link them, you need to do this, https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/java-plugin-development-guide/#contextsnapshot

**Reply by someone-1** [reply]
thanks for suggestion @wu-sheng will understand this
